accord-core/src/main/java/accord/local/Node.java

```diff
@@ -458,8 +485,13 @@ public void send(Id to, Request send)
 
     public void reply(Id replyingToNode, ReplyContext replyContext, Reply send)
     {
+        // TODO (usability, now): add Throwable as an argument so the error check is here, every single message gets this wrong causing a NPE here
         if (send == null)
-            throw new NullPointerException();
+        {
+            NullPointerException e = new NullPointerException();
+            agent.onUncaughtException(e);
+            throw e;
```

Comment:
If the exception is thrown should it really go the uncaught route? Also don't quite follow the TODO

Reply:
> If the exception is thrown should it really go the uncaught route?

I was seeing `BurnTest` failing in these cases, so something is catching and not-handling this NPE... which makes `BurnTest` pass!  With this change these NPEs cause `BurnTest` to fail right away

Reply:
Hmm yeah I have seen a lot of `null` messages ending up here as well. The error handling outside.

Better reported twice than not reported at all for now.